components/blocks/nav/SiteNavbar/SiteNavbar.test.ts

```diff
@@ -0,0 +1,94 @@
+import { ref } from 'vue'
+import { useState } from '#app'
+import SiteNavbar from './SiteNavbar.vue'
+import * as apiComposables from '@/composables/api'
+import { User } from '@/lib/api/data-contracts'
+import { fireEvent, stubbedRender } from '@/testUtils'
+
+vi.mock('#app', () => ({
+  useState: vi.fn(),
+}))
+
+afterEach(() => {
+  vi.restoreAllMocks()
+  vi.clearAllMocks()
+})
+
+describe('<SiteNavbar />', () => {
+  it('should render without crashing', () => {
+    useState.mockImplementation(() => ref())
+
+    const { unmount } = stubbedRender(SiteNavbar)
+
+    unmount()
+  })
+
+  describe('when no user is logged in', () => {
+    beforeEach(() => {
+      useState.mockImplementationOnce((_stateId: string) =>
```

Comment:
Nit: could just be `mockImplementation` here right? I don't see any further mocking of this in the rest of the tests